pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -389,7 +389,22 @@ func (p *PriorityQueue) isPodWorthRequeuing(logger klog.Logger, pInfo *framework
 				continue
 			}
 
-			switch h := hintfn.QueueingHintFn(logger, pod, oldObj, newObj); h {
+			h, err := hintfn.QueueingHintFn(logger, pod, oldObj, newObj)
+			if err != nil {
+				// If the QueueingHintFn returned an error, we should treat the event as QueueAfterBackoff so that we can prevent
+				// the Pod from stucking in the unschedulable pod pool.
+				var oldObjRef, newObjRef klog.ObjectRef
```

Comment:
@carlory 

Ah, so we cannot use interface{} in `klog.KObj`...
Cannot we somehow utilize As function here to convert two objects more wisely?

Reply:
@sanposhiho 

```go
func As[T runtime.Object](oldObj, newobj interface{}) (T, T, error)
```

I'm trying to use this function to convert the oldObj and newObj to the klog.KMetadata type. But it doesn't work.

The As function only convert the oldObj and newObj to the runtime.Object type. And I cannot find any existed function to convert the interface{} to the klog.KMetadata type.

Reply:
After change the func to 

```go
func As[T any](oldObj, newobj interface{}) (T, T, error)
```

It works.

Reply:
After change the func to 

```go
func As[T any](oldObj, newobj interface{}) (T, T, error)
```

It works.